### DISCUSSION TITLE
v3 Modern Search - Sort Issue

### DISCUSSION THREAD

**Discussion by manojv1980** [microsoft-search/pnp-modern-search#3600] [Q&A]
Hi,
I'm using v3 Results webpart to display all pdf documents of a particular content type on a modern Sharepoint page in SP Online. This is the search query on first page of the webpart:

`Path:{SiteCollection.URL} ContentType:"Marketing Docs" FileType=PDF RefinableString10:"English" -RefinableString10:"English (US)" RefinableString13="Released" `

The refinable strings are used to filter by language (metadata for marketing documents) and whether document is "Released" or "Obsolete".

_Results Layout_ is selected as _Details list_ and displaying _Title_ and _Created_ fields by selecting them under _Managed Columns_. Everything else is default settings. 

Issue is with the above setting I can click on the _Title_ or _Created_ column header and sort but if I add _Sortable properties_ using _Edit sortable fields_, then sort doesn't work when I select say _Title_ in the Sort dropdown, not descending or ascending sort.
Is this a known issue? I'm adding screenshots of my webpart config. 
<img width="640" alt="1" src="https://github.com/microsoft-search/pnp-modern-search/assets/162339924/82e268f9-823b-4b53-8bf2-cc48ad3c55b8">
<img width="188" alt="2" src="https://github.com/microsoft-search/pnp-modern-search/assets/162339924/69c5e2c9-86e9-4a08-9784-c0a84f7a92a7">
<img width="548" alt="3" src="https://github.com/microsoft-search/pnp-modern-search/assets/162339924/5716a61a-2171-4d4b-a39a-0a53520216d6">

I didn't try v4 because I read sort dropdown control isn't yet implemented in the Results webpart.

**Comment by kasperbolarsen** [comment]
We can only support the current release which is 4.10* (soon 4.11) 

Title is by default not a sortable managed property, but you should be able to find a guide in the discussions on how to connect Title to a RefinableString and thereby making it sortable.

**Reply by manojv1980** [reply]
Thank you for the reply.

I did try mapping crawled properties  _ows_Title_, _Title_, _MetadataExtractorTitle_, _TermTitle_, _ows_BaseName_, _Office:#2_ to a _RefinableString_, did a site reindex but that didn't make any difference - sort wasn't working.

> <img width="715" alt="5" src="https://github.com/microsoft-search/pnp-modern-search/assets/162339924/5a154e70-60d5-4e7f-8da4-bedb365dc136">

I wanted to sort by _LastModifiedDate_ as well so mapped crawled property _ows_modifed_ to a _RefinableDate_ managed property, did a reindex of the site, but again sort wasn't working as expected.

> <img width="645" alt="4" src="https://github.com/microsoft-search/pnp-modern-search/assets/162339924/cc1e8d25-6776-4114-b7cd-796718ead007">

I can understand you wouldn't support an older version but from what I read the newer version doesn't support sort fields dropdown as a functionality yet, hence still using v3. We wouldn't have gone with an open source webpart if Microsoft had provided similar functionality in their Search results webpart.

I am using this v3 version of Results webpart without any customisation or any changes, just in vanilla state.

**Reply by wobba** [reply]
Also depends where you did the mapping (site schema, tenant schema) and where you are searching from. These types of mappings should be done at the tenant level to ensure they work (in my experience). And you don't know when re-indexing has happened either unless you check the crawl log. As it can take up to 72h in some cases.

**Reply by manojv1980** [reply]
@wobba Thank you the reply.
The mappings were done at the Tenant level and I have checked using SP Search Query Tool v2.10.0 and can see my managed properties have values so not sure why this isn't working with sort.

I also tried using a refinablestring managed property with sort that was configured years back and even with this the sort on v3 webpart isn't working. 
However, if I create a Result Source on the Site with the same query and select my new managed property for sort, it seems to work in there.

**Reply by wobba** [reply]
Weird behavior indeed. And logging API bugs is unfortunately tricky.
